File: assetforge-frontend/src/types/Ticket.ts
export interface Ticket {
    id: number;
    ticketNumber: string;
    prefix: string;
    title: string;
    description: string;
    status: string;
    priority: string;
    createdAt: string;

    // Who created the ticket
    createdById: number;
    createdByName?: string;

    // Who is it assigned to
    assignedToId?: number;
    assignedToName?: string;
}

File: AssetForge.Infrastructure/Entities/Ticket.cs
namespace AssetForge.Infrastructure.Entities
{
    public class Ticket
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Status { get; set; } = "Open"; // Open, InProgress, Closed
        public string Priority { get; set; } = "Medium"; // Low, Medium, High
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Creator
        public int CreatedById { get; set; }
        public User CreatedBy { get; set; }

        // Assigned
        public int? AssignedToId { get; set; }
        public User? AssignedTo { get; set; }

        // Ticket number
        public string? Prefix { get; set; } = "TKT";
        public string TicketNumber { get; set; } = string.Empty;
    }
}
